src/include/catalog/catalog_defs.h

```diff
@@ -39,6 +39,11 @@ constexpr trigger_oid_t INVALID_TRIGGER_OID = trigger_oid_t(NULL_OID);
 constexpr type_oid_t INVALID_TYPE_OID = type_oid_t(NULL_OID);
 constexpr view_oid_t INVALID_VIEW_OID = view_oid_t(NULL_OID);
 
+constexpr uint32_t TEMP_OID_MASK = 1 << 31;
+#define TEMP_OID(strongtype, x) (strongtype(x | catalog::TEMP_OID_MASK))
```

Comment:
I think we strongly prefer functions over macros, unless there is a good reason for doing this.

Reply:
As I think was discussed with Tanuj, this will be staying as a macro for the time being

Reply:
Can you document this conversation? Because my immediate response was scrolling by this was "Get rid of that macro."